pkg/services/preference/prefimpl/store_test.go

```diff
@@ -0,0 +1,165 @@
+				TeamID:          2,
+				Created:         time.Now(),
+				Updated:         time.Now(),
+			})
+		require.NoError(t, err)
+
+		query := &pref.Preference{OrgID: 1, UserID: 1, TeamID: 2}
+		prefs, err := prefStore.Get(context.Background(), query)
+		require.NoError(t, err)
+		require.Equal(t, int64(4), prefs.HomeDashboardID)
+	})
+
+	t.Run("List with saved org and user home dashboard should return user home dashboard", func(t *testing.T) {
+		_, err := prefStore.Insert(context.Background(),
+			&pref.Preference{
+				OrgID:           1,
+				UserID:          1,
+				TeamID:          3,
+				HomeDashboardID: 1,
+				Created:         time.Now(),
+				Updated:         time.Now(),
+			})
+		require.NoError(t, err)
+
+		query := &pref.Preference{OrgID: 1, UserID: 1, Teams: []int64{2}}
+		prefs, err := prefStore.List(context.Background(), query)
+		require.NoError(t, err)
+		require.Equal(t, int64(4), prefs[0].HomeDashboardID)
+	})
+
+	t.Run("List with saved org and other user home dashboard should return org home dashboard", func(t *testing.T) {
+		_, err := prefStore.Insert(context.Background(),
+			&pref.Preference{
+				OrgID:           1,
+				UserID:          2,
+				TeamID:          3,
+				HomeDashboardID: 1,
+				Created:         time.Now(),
+				Updated:         time.Now(),
+			})
+		require.NoError(t, err)
+
+		query := &pref.Preference{OrgID: 1, UserID: 1, Teams: []int64{3}}
+		prefs, err := prefStore.List(context.Background(), query)
+		require.NoError(t, err)
+		require.Equal(t, int64(1), prefs[0].HomeDashboardID)
+		require.Equal(t, int64(1), prefs[1].HomeDashboardID)
+	})
+
+	t.Run("List with saved org and teams home dashboard should return last team home dashboard", func(t *testing.T) {
+		query := &pref.Preference{
+			OrgID: 1, Teams: []int64{2, 3},
+		}
+		prefs, err := prefStore.List(context.Background(), query)
+		require.NoError(t, err)
+		require.Equal(t, int64(4), prefs[0].HomeDashboardID)
+		require.Equal(t, int64(1), prefs[1].HomeDashboardID)
+		require.Equal(t, int64(1), prefs[2].HomeDashboardID)
+	})
+
+	t.Run("List with saved org and other teams home dashboard should return org home dashboard", func(t *testing.T) {
+		_, err := prefStore.Insert(context.Background(), &pref.Preference{OrgID: 1, HomeDashboardID: 1, Created: time.Now(), Updated: time.Now()})
+		require.NoError(t, err)
+		_, err = prefStore.Insert(context.Background(), &pref.Preference{OrgID: 1, TeamID: 2, HomeDashboardID: 2, Created: time.Now(), Updated: time.Now()})
+		require.NoError(t, err)
+		_, err = prefStore.Insert(context.Background(), &pref.Preference{OrgID: 1, TeamID: 3, HomeDashboardID: 3, Created: time.Now(), Updated: time.Now()})
+		require.NoError(t, err)
+
+		query := &pref.Preference{OrgID: 1}
+		prefs, err := prefStore.List(context.Background(), query)
+		require.NoError(t, err)
+		require.Equal(t, int64(1), prefs[0].HomeDashboardID)
+	})
+
+	t.Run("Update for a user should only modify a single value", func(t *testing.T) {
```

Comment:
this fails with PostgreSQL for some reason:
```
=== RUN   TestPreferencesDataAccess/Update_for_a_user_should_only_modify_a_single_value
    store_test.go:155: Result mismatch (-want +got):
          &pref.Preference{
                ... // 7 identical fields
                Timezone:  "browser",
                WeekStart: "1",
        -       Theme:     "light",
        +       Theme:     "dark",
                Created:   s"2022-04-14 08:27:05 +0000 UTC",
                Updated:   s"2022-04-14 08:27:05 +0000 UTC",
                JSONData:  &{Navbar: {SavedItems: {{ID: "alerting", Text: "Alerting", Url: "/alerting"}}}},
          }

I run it using:
```
GRAFANA_TEST_DB=postgres go test -v ./pkg/services/preference/prefimpl/... -tags=integration -run TestPreferencesDataAccess
```

Reply:
oh yes, thanks Sofia